server-actions/user-actions.ts

```diff
@@ -21,25 +21,62 @@ export async function createUser(email: string) {
   });
 }
 
-export async function updateUserRewards(email: string, room_cost: number) {
-  try {
-    const pointsToAdd = room_cost * DEFAULT_REWARDS_MULTIPLIER;
+export async function getUserRewards(email: string) {
+  const user = await prisma.user.findUnique({
+    where: { email },
+    select: { rewardPoints: true },
+  });
 
-    const updatedRewards = await prisma.user.update({
-      where: {
-        email: email,
-      },
-      data: {
-        rewardPoints: {
-          increment: pointsToAdd,
-        },
-      },
-    });
+  if (!user) {
+    throw new Error("User not found");
+  }
+
+  return user.rewardPoints;
+}
 
-    console.log("Updated rewards:", updatedRewards);
-    return updatedRewards;
-  } catch (error: any) {
-    console.error("Error updating rewards:", error.message);
-    throw error;
+export async function updateUserRewards(email: string, payment: number) {
+  const user = await prisma.user.findUnique({
+    where: { email },
+    select: { rewardPoints: true },
+  });
+
+  if (!user) {
+    throw new Error("User not found");
   }
+  const oldPoints = user.rewardPoints;
+
+  const rewardPoints = oldPoints ?? 0 + payment * DEFAULT_REWARDS_MULTIPLIER;
+
+  return prisma.user.update({
+    where: { email },
+    data: { rewardPoints },
+  });
```

Comment:
```suggestion

  const additionalRewardPoints = payment * DEFAULT_REWARDS_MULTIPLIER;

  return prisma.user.update({
    where: { email },
    data: {
      rewardPoints: {
        increment: additionalRewardPoints,
      },
    },
  });
```

Reply:
I wasn't sure if this would work with negatives. The docs are unclear, but it looks like it works. Note that this function is intended to support all updates (negative and positive). This way it can be used in cancellation as well.